go/p2p/rpc/peermgmt.go

```diff
@@ -115,6 +116,26 @@ type PeerManager interface {
 	// GetBestPeers returns a set of peers sorted by the probability that they will be able to
 	// answer our requests the fastest with some randomization.
 	GetBestPeers(opts ...BestPeersOption) []core.PeerID
+
+	// WatchUpdates returns a channel that produces a stream of messages on peer updates.
+	WatchUpdates() (<-chan PeerUpdate, pubsub.ClosableSubscription, error)
+}
+
+// PeerUpdate is a peer update event.
+type PeerUpdate struct {
+	ID core.PeerID
```

Comment:
Not sure if the intention is for these structures to be serializable? But if so, for consistency this should have an annotation to rename the field to `id`.

Reply:
no need for serialization at the moment, so I removed the existing struct tags.